### PR TITLE
add dnd-character exercise

### LINKED ISSUE #667: Implement new exercise: dnd-character
There is a new exercise, [dnd-character](https://github.com/exercism/problem-specifications/blob/master/exercises/dnd-character/description.md), which data can be found here: https://github.com/exercism/problem-specifications/tree/master/exercises/dnd-character

To implement the `dnd-character` exercise, the following needs to be done:

- Create a directory for the new exercise in the `exercises` directory (note: this has to match the exact exercise name).
- Add a new entry to the [config.json](https://github.com/exercism/fsharp/blob/master/config.json) file.
- Create a generator to automatically convert the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/dnd-character/canonical-data.json) to a test file. For more information on how to do this, check the [generators docs](https://github.com/exercism/fsharp/blob/master/docs/GENERATORS.md).
- Create a stub implementation file that contains the code necessary to compile (which are the methods called by the test file), but lacks an implementation. (see [this example](https://github.com/exercism/fsharp/blob/master/exercises/two-fer/TwoFer.cs)).
- Create an implementation file that passes all the tests.
- Create a project file for the project (you can probably best copy and modify one of the existing projects).

Note that this exercise should be implemented using property-based tests where possible. An example of an existing exercises that uses property-based tests is the [diamond](https://github.com/exercism/fsharp/blob/master/exercises/diamond/DiamondTest.fs) exercise.

### PR DESCRIPTION
**Pull request by davidelettieri**
I decided to follow the C# example.

closes #667

### CODE REVIEW

File: exercises/dnd-character/DndCharacter.fs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿module DndCharacter
+
+open System
+
+let modifier x =
```
Comment (ErikSchierboom) [inline exercises/dnd-character/DndCharacter.fs]:
```suggestion
let modifier score =
```

Perhaps use a more descriptive name than `x`?

File: exercises/dnd-character/DndCharacterTest.fs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    let ability = ability()
+    for i in 1 .. 10 do
+        ability |> should be (greaterThanOrEqualTo 3)
+        ability |> should be (lessThanOrEqualTo  18)
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Random character is valid`` () =
+    for i in 1 .. 10 do
+        let character = DndCharacter()
+        character.strength |> should be (greaterThanOrEqualTo 3)
```
Comment (ErikSchierboom) [inline exercises/dnd-character/DndCharacterTest.fs]: The default convention is to use PascalCase for members: https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#use-pascalcase-for-type-declarations-members-and-labels Could you convert the member names to PascalCase?

**Comment by ErikSchierboom** [comment]
Awesome!

**Comment by ErikSchierboom** [comment]
@davidelettieri Your contributions have been of a consistently high level. Are you perhaps interested in becoming a maintainer of the F# track? This would mean that you get the ability to merge PR's and help shape the F# track. In practice, you could choose what you'd like to work on. If you'd like to focus on creating new exercises: fine! If you'd like to focus on reviewing other PR's, that's great too.

Feel free to let me know if you're not interested. No harm will be done :)

**Comment by davidelettieri** [comment]
Ehi @ErikSchierboom I would love to! What kind of committment is expected to be a mantainer?

**Comment by ErikSchierboom** [comment]
To be honest: none :) It really depends on you, how much time _you_ want to spend. If you are very busy and are only interested to do the occasional review, that is totally fine! If you'd like to do more extensive work (e.g. adding new exercises and such), that is of course also more than welcome! For a more complete list of all the things you can do, see the [Become a Maintainer](https://exercism.io/become-a-maintainer) page.

Shall I start the process of getting you added as a maintainer?

**Comment by davidelettieri** [comment]
Sure and thanks for asking! :)

**Comment by ErikSchierboom** [comment]
@kytrinyx @iHiD What are the steps for @davidelettieri to become an F# maintainer?

**Comment by kytrinyx** [comment]
@ErikSchierboom The first thing to try is to have them add themselves to the `config/maintainers.json` file. We have a bot that _sometimes_ works 😂 that will invite them to the team.

If that doesn't work, then please ping me and I will do it manually :)

**Comment by ErikSchierboom** [comment]
Thanks!

@davidelettieri Could you add yourself as a maintainer to https://github.com/exercism/fsharp/blob/master/config/maintainers.json?

**Pull request merged**